x/evidence/keeper/infraction.go

```diff
@@ -44,15 +39,28 @@ func (k Keeper) HandleDoubleSign(ctx sdk.Context, evidence types.Equivocation) {
 		return
 	}
 
-	// reject evidence if the double-sign is too old
-	if age > k.MaxEvidenceAge(ctx) {
-		logger.Info(
-			fmt.Sprintf(
-				"ignored double sign from %s at height %d, age of %d past max age of %d",
-				consAddr, infractionHeight, age, k.MaxEvidenceAge(ctx),
-			),
-		)
-		return
+	// calculate the age of the evidence
+	infractionHeight := evidence.GetHeight()
+	infractionTime := evidence.GetTime()
+	ageDuration := ctx.BlockHeader().Time.Sub(infractionTime)
+	ageBlocks := ctx.BlockHeader().Height - infractionHeight
+
+	// Reject evidence if the double-sign is too old. Evidence is considered stale
+	// if the differenc in time and number of blocks is greater than the allowed
+	// parameters defined.
+	cp := ctx.ConsensusParams()
+	if cp != nil && cp.Evidence != nil {
+		if ageDuration > cp.Evidence.MaxAgeDuration && ageBlocks > cp.Evidence.MaxAgeNumBlocks {
+			logger.Info(
+				"ignored equivocation; evidence too old",
+				"validator", consAddr,
+				"infraction_height", infractionHeight,
+				"max_age_num_blocks", cp.Evidence.MaxAgeNumBlocks,
+				"infraction_time", infractionTime,
+				"max_age_duration", cp.Evidence.MaxAgeDuration,
+			)
+			return
+		}
 	}
```

Comment:
This is the core change with regard to the initial intent of the PR.

/cc @cwgoes @ebuchman @melekes